bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/SaveManager.java

```diff
@@ -1508,8 +1508,9 @@ protected void snapTree(ElementTree tree, IProgressMonitor monitor) throws CoreE
 	 *              ElementTreess have the newest ElementTree as ancestor
 	 *              (transitive parent). The given trees do not need to contain all
 	 *              ElementTrees from the getParent() relationship.
-	 * @return trees ordered by ElementTree.treeStamp descending. i.e. newest
-	 *         ElementTree (without ancestor within the trees) first.
+	 * @return null or trees ordered by ElementTree.treeStamp descending. i.e.
+	 *         newest ElementTree (without ancestor within the trees) first. Returns
+	 *         null when the preconditions for sorting are not met.
```

Comment:
Could it really return null? You are throwing an exception.

Reply:
I was looking at the wrong file.

Reply:
welcome to github's confusing presentation: **That** method does not throw but logs an NPE